fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaDataBuilder.java

```diff
@@ -919,14 +1024,9 @@ public RecordMetaData build(boolean validate) {
                     index.setPrimaryKeyComponentPositions(buildPrimaryKeyComponentPositions(index.getRootExpression(), primaryKey));
                 }
             } else {
-                for (Index index : recordTypeBuilder.getIndexes()) {
-                    if (logger.isWarnEnabled()) {
-                        logger.warn(KeyValueLogMessage.of("Created Index indexes a record type without a primary key",
-                                "record_type", recordTypeBuilder.getName(),
-                                "indexName", index.getName()));
-                    }
-                    indexes.remove(index.getName());
-                }
+                throw new MetaDataException("Record type " +
+                                            recordTypeBuilder.getName() +
+                                            " must have a primary key");
```

Comment:
I see there were a bunch of tests changed to make it so that we handle the fact that things might not have a primary key. Are there any tests were we validate that a record type without a primary key throws an error?